server.js

```diff
@@ -26,7 +80,94 @@ app.use(express.json());
 
 // Start defining your routes here
 app.get("/", (req, res) => {
-  res.send("Hello Technigo!");
+  res.status(200).json({
+    Message: "Top 50 songs - data",
+    data: listEndpoints(app)
+  });
+});
+
+app.get("/songs", async (req, res) => {
+  const songs = await Song.find().sort({id: 1});
+  res.status(200).json({
+    success: true,
+    message: "OK",
+    body: {
+      topMusicData: songs
+    }
+  });
+});
+
+app.get("/songsOnlyTitleAndArtist", async (req, res) => {
+  const songsOnlyTitleAndArtist = await Song.find().sort({id: 1}).select({trackName: 1, artistName: 1})
+  res.status(200).json({
+    success: true,
+    message: "OK",
+    body: {
+      topMusicData: songsOnlyTitleAndArtist
+    }
+  });
+});
+
+app.get('/song/id/:id', async (req, res) => {
+  try {
+    const songById = await Song.findById(req.params.id);
+    if (songById) {
+      res.status(200).json({
+        success: true,
+        message: "OK",
+        body: {
+          topMusicData: songById
+        }
+      });
+    } else {
+      res.status(404).json({
+        success: false,
+        message: "Song Not found",
+        body: {
+          topMusicData: {}
+        }
+      });
+    }
+  } catch (error) {
+    res.status(400).json({
+      success: false,
+      message: "Invalid id",
+      body: {
+        topMusicData: {}
+      }
+    });
+  }
+})
```

Comment:
This endpoint works with findById, it shows the correct data. However, I could not get to error message : songs not found when I entered the wrong ID. It went straight to error 400 right away.